From 1762a8f3ef26de2617108c2e184a7cebcfc7525b Mon Sep 17 00:00:00 2001
From: "Valery V. Vorotyntsev" <valery_vorotyntsev@xyratex.com>
Date: Wed, 20 Mar 2013 23:30:52 +0200
Subject: [PATCH 48/63] rpc-ub: abandon the idea of configurable xprt

SERVER_ENDPOINT is used in static g_argv array. Making it dependent
on run-time parameter is too much hassle.
---
 rpc/ub/ub.c | 29 +++++------------------------
 1 file changed, 5 insertions(+), 24 deletions(-)

diff --git a/rpc/ub/ub.c b/rpc/ub/ub.c
index a97bb43..b7c7659 100644
--- a/rpc/ub/ub.c
+++ b/rpc/ub/ub.c
@@ -41,8 +41,7 @@
 struct args {
 	unsigned int a_nr_conns;
 	unsigned int a_nr_slots;
-	unsigned int a_msg_len;
-	bool         a_use_lnet;
+	unsigned int a_msg_len; /* XXX USEME */
 };
 
 /** Assigns default values to the arguments. */
@@ -51,7 +50,6 @@ static void args_init(struct args *args)
 	args->a_nr_conns = 1;
 	args->a_nr_slots = 15;
 	args->a_msg_len  = 32;
-	args->a_use_lnet = false;
 }
 
 enum { /* XXX TODO: Revise these values. */
@@ -63,7 +61,7 @@ enum { /* XXX TODO: Revise these values. */
 static int args_format_error(void)
 {
 	fprintf(stderr, "**ERROR** Invalid format of the string with options.\n"
-		"  Expected: [lnet,]nr_conns=NUM,nr_slots=NUM,msg_len=NUM\n");
+		"  Expected: nr_conns=NUM,nr_slots=NUM,msg_len=NUM\n");
 	return -EINVAL;
 }
 
@@ -86,25 +84,8 @@ static int args_parse(const char *src, struct args *dest)
 {
 	if (src == NULL || *src == 0)
 		return 0;
-
-	int rc;
-
-	rc = strncmp(src, "lnet", 4);
-	if (rc == 0) {
-		src += 4;
-		if (*src == 0) {
-			dest->a_use_lnet = true;
-			return 0;
-		} else if (*src == ',') {
-			dest->a_use_lnet = true;
-			++src;
-		} else {
-			return args_format_error();
-		}
-	}
-
-	rc = sscanf(src, "nr_conns=%u,nr_slots=%u,msg_len=%u",
-		    &dest->a_nr_conns, &dest->a_nr_slots, &dest->a_msg_len);
+	int rc = sscanf(src, "nr_conns=%u,nr_slots=%u,msg_len=%u",
+			&dest->a_nr_conns, &dest->a_nr_slots, &dest->a_msg_len);
 	return rc == 3 ? args_check_limits(dest) : args_format_error();
 }
 
@@ -121,7 +102,7 @@ enum {
 	MAX_RETRIES        = 5
 };
 
-/* #define UB_USE_LNET_XPORT  /\* XXX CONFIGUREME *\/ */
+/* #define UB_USE_LNET_XPORT */
 #ifdef UB_USE_LNET_XPORT
 #  define CLIENT_ENDPOINT_FMT  "0@lo:12345:34:%d"
 #  define SERVER_ENDPOINT_ADDR "0@lo:12345:12:1"
-- 
1.8.3.2

